Remotion/Web/Development.WebTesting/WebTestSetUpFixtureHelper.cs

```diff
@@ -144,31 +144,37 @@ private void VerifyWebApplicationStarted (Uri webApplicationRoot, TimeSpan appli
 
       var stopwatch = Stopwatch.StartNew();
 
-#pragma warning disable SYSLIB0014
-      var webRequest = (HttpWebRequest)HttpWebRequest.Create(resolvedUri); // TODO RM-8492: Replace with HttpClient
-#pragma warning restore SYSLIB0014
-      webRequest.Method = WebRequestMethods.Http.Head;
-      webRequest.AllowAutoRedirect = true;
-      webRequest.Host = webApplicationRoot.Host;
-      webRequest.ServerCertificateValidationCallback += (sender, certificate, chain, errors) => true;
-
       HttpStatusCode statusCode = default;
       Assertion.DebugAssert(statusCode != HttpStatusCode.OK);
 
       while (statusCode != HttpStatusCode.OK)
       {
+  #pragma warning disable SYSLIB0014
+        var webRequest = (HttpWebRequest)HttpWebRequest.Create(resolvedUri); // TODO RM-8492: Replace with HttpClient
+  #pragma warning restore SYSLIB0014
+        webRequest.Method = WebRequestMethods.Http.Head;
+        webRequest.AllowAutoRedirect = true;
+        webRequest.Host = webApplicationRoot.GetComponents(UriComponents.Host | UriComponents.Port, UriFormat.UriEscaped);
```

Comment:
Do we need to specify UriFormat.UriEscaped at this point given we only use the existing hostname? I don't mind being safe, but if it's just overprotectiveness, it may reduce readability.

Reply:
We have to supply a value, there is no overload with only one parameter and this is the default that is used in URI in other formatting scenarios.